gsuite_base/eg_common_api.e

```diff
@@ -0,0 +1,359 @@
+					logger.write_error ("parse_last_response->Unauthorized status, review your authorization credentials")
+				end
+				if attached l_response.body as l_body then
+					logger.write_debug ("parse_last_response->body:" + l_body)
+					create l_json_parser.make_with_string (l_body)
+					l_json_parser.parse_content
+					if l_json_parser.is_valid then
+						if attached {JSON_OBJECT} l_json_parser.parsed_json_value as l_main_jso then
+							if attached {JSON_OBJECT} l_main_jso.item ("error") as l_error_jso then
+								if attached {JSON_NUMBER} l_error_jso.item ("code") as l_jso then
+									print ("parse_last_response-> error code:" + l_jso.representation + "%N")
+								end
+								if attached {JSON_STRING} l_error_jso.item ("message") as l_jso then
+									print ("parse_last_response-> error message:" + l_jso.unescaped_string_8 + "%N")
+								end
+								if attached {JSON_STRING} l_error_jso.item ("status") as l_jso then
+									print ("parse_last_response-> error status:" + l_jso.unescaped_string_8 + "%N")
+								end
+							end
+						end
+					else
+						print ("parse_last_response-> Error: Invalid json body content:" + l_body + "%N")
+					end
+				end
+			end
+		end
+
+	has_error: BOOLEAN
+			-- Last api call raise an error?
+		do
+			if attached last_response as l_response then
+				Result := l_response.status /= 200
+			else
+				Result := False
+			end
+		end
+
+	error_message: STRING
+			-- Last api call error message.
+		require
+			has_error: has_error
+		do
+			if
+				attached last_response as l_response
+			then
+				if
+					attached l_response.error_message as l_error_message
+				then
+					Result := l_error_message
+				else
+					Result := l_response.status.out
+				end
+			else
+				Result := "Unknown Error"
+			end
+		end
+
+feature {NONE} -- Implementation
+
+	api_post_call (a_api_url: STRING; a_query_params: detachable STRING_TABLE [STRING]; a_payload: detachable STRING; a_upload_data: detachable TUPLE[data:PATH; content_type: STRING])
+		note
+			eis: "name=payload_body", "src=https://tools.ietf.org/html/draft-ietf-httpbis-p3-payload-14#section-3.2", "protocol=https://tools.ietf.org/html/draft-ietf-httpbis-p3-payload-14#section-3.2"
+			-- POST REST API call for `a_api_url'
+		do
+			internal_api_call (a_api_url, "POST", a_query_params, a_payload, a_upload_data)
+		end
+
+	api_delete_call (a_api_url: STRING; a_query_params: detachable STRING_TABLE [STRING])
+			-- DELETE REST API call for `a_api_url'
+		do
+			internal_api_call (a_api_url, "DELETE", a_query_params, Void, Void)
+		end
+
+	api_get_call (a_api_url: STRING; a_query_params: detachable STRING_TABLE [STRING])
+			-- GET REST API call for `a_api_url'
+		do
+			internal_api_call (a_api_url, "GET", a_query_params, Void, Void)
+		end
+
+	internal_api_call (a_api_url: STRING; a_method: STRING; a_query_params: detachable STRING_TABLE [STRING]; a_payload: detachable STRING; a_upload_data: detachable TUPLE[filename:PATH; content_type: STRING])
```

Comment:
Maybe you should elaborate a little regarding this feature. I am not ure about the purpose of this and the feature name does not say much